### PR TITLE
Relationships-pull-request

### PR DESCRIPTION
**Pull request by ikoote1**
Changes made on the code:
1. Created a class Classroom with the following:

- @label instance variable, should be initialized in the constructor.
- Setter and getter for @label (remember about attr_accessor).

2. Created the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

- Created the has-many side (a classroom has many students).
- Created the belongs-to side (a student belongs to a classroom).
- Made sure that when adding a student to a classroom it also sets the classroom for the student.
- Made sure that when setting the classroom for a student it also adds it to the classrooms' students.

3. Created a class Book with the following:

- @title and @author instance variables, should be initialized in the constructor.
- Setters and getters for instance variables (remember about attr_accessor). 

4. Created a class Rental with the following:

- @date instance variable, should be initialized in the constructor.
- Setter and getter for @date (remember about attr_accessor). 

5. Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

- Created the has-many side of Book and Rental (a book has many rentals).
- Created the belongs-to side of Rental and Book (a rental belongs to a book).
- Created the has-many side of Person and Rental (a person has many rentals).
- Created the belongs-to side of Rental and Person (a rental belongs to a person).
- Modified the constructor of Rental so Book and Person are set in it.

6. Opened a pull request.

Thanks for reviewing my code.

### CODE REVIEW

File: modules/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Classroom
+  attr_accessor :label, :students
+
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students << student
```
Comment (Ismailco) [inline modules/classroom.rb]: - [x] Kindly make sure to add the classroom to the student.classroom e.g `student.classroom = self`

File: modules/book.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
```
Comment (Ismailco) [inline modules/book.rb]: - [x] Kindly make sure to use the `Rental` class to create a new rental in this method e.g `Rental.new(date, self, person)`

File: modules/rental.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date)
```
Comment (Ismailco) [inline modules/rental.rb]: - [x] Kindly consider adding the `book` and `person` as parameters for the constructor method

File: modules/rental.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    @date = date
+    @book = book
+    @person = person
```
Comment (Ismailco) [inline modules/rental.rb]: - [x] Nice work creating @book and @person, however, The `rental` should add itself to the `@book` and @person` achieving the 'belongs-to' association

```suggestion
    @book = book
    @person = person
    book.rentals << self
    person.rentals << self
```

File: modules/person.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
+require_relative './nameable'
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown')
```
Comment (Ismailco) [inline modules/person.rb]: - [x] Nice work here however you should add `parrent_permition` as a parameter for the constructor method

File: modules/person.rb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  attr_accessor :name, :age, :rentals
+  attr_reader :id
+
+  def correct_name
+    name
+  end
+
+  def rent_book(rental)
+    @rentals << rental
+  end
```
Comment (Ismailco) [inline modules/person.rb]: - [x] Same for this method kindly consider using the ` Rentals` class to create a new rental 

```suggestion
  def rent_book(rental)
    Rental.new(date, book, self)
  end
```

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @ikoote1, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: modules/person.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
+require_relative './nameable'
+
+class Person < Nameable
```
Comment (OyePriscilla) [inline modules/person.rb:3]: - [ ] As suggested earlier and according to the project requirement, kindly implement a method to create a new rental using the `Rentals` class.

 ```
def rent_book(book, date)
    Rental.new(date, book, self)
  end
```

File: modules/book.rb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+class Book
```
Comment (OyePriscilla) [inline modules/book.rb:1]: - [ ] As suggested earlier and according to the project requirement, kindly implement a method to create a new rental using the Rentals class.

```` 
def add_rental (person, date)
    Rental.new(date, self, person)
  end
````

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @ikoote1 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Set up associations between classes and objects ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by frank1738 — APPROVED** [review]
Hi @ikoote1 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**